netmon.cc

```diff
@@ -467,13 +486,18 @@ CheckResult PINGChecker::perform()
 
     SConnect(sock, s);
     SSetsockopt(sock, SOL_IP, IP_RECVTTL, 1);
+    if (d_dontFragment) {
+      SSetsockopt(sock, IPPROTO_IP, IP_MTU_DISCOVER, IP_PMTUDISC_DO);
+    } else {
+      SSetsockopt(sock, IPPROTO_IP, IP_MTU_DISCOVER, IP_PMTUDISC_DONT);
+    }
 
-    string packet = makeICMPQuery(s.sin4.sin_family, 1, 1);
+    string packet = makeICMPQuery(s.sin4.sin_family, 1, 1, d_size);
     DTime dt;
     dt.start();
     SWrite(sock, packet);
-    double timeo=1.0;
-    if(!waitForData(sock, &timeo)) { // timeout
+
+    if(!waitForData(sock, &d_timeout)) {
```

Comment:
one day waitForTimeout might start decrementing that timeout value, so best to keep this as 'double timeo = d_timeout'

Reply:
Amended sources, adjusted as you suggested.